### ISSUE TITLE
Only the first before and after function executed

### ISSUE THREAD

**Issue by Tevek** [timjroberts/cucumber-js-tsflow#109]
If you have to use multiple before and after methods within hooks it seems only the first before(), after() executed in Version 4.0.0-rc.9
regardless any tag

Below 4.0.0-rc.9 the before,after does not apply the HookOption as parameter so we are not able to use individual timeouts for each hook function

**Comment by Fryuni** [comment]
Hey @Tevek, thank you for reporting this issue!

I'm a bit confused so, so be clear, do you mean this?
```ts
@binding()
class Foo {
  @before()
  public beforeOne() {
    // This runs
  }

  @before()
  public beforeTwo() {
    // This doesn't
  }

  @after()
  public afterOne() {
    // This runs
  }

  @after()
  public afterTwo() {
    // This doesn't
  }
}
```

I don't get where timeouts play a role in what you are reporting, can you send a snippet of a case where it is breaking?

**Comment by Tevek** [comment]
Yes. correct
beforeOne and afterOne run only

Timeout is a different issue do not deal with it

**Comment by Fryuni** [comment]
This was fixed on `4.0.0-rc.11`
